src/util/triplea/MapXMLCreator/GameSettingsRow.java

```diff
@@ -0,0 +1,369 @@
+		m_tEditable.addFocusListener(new FocusListener()
+		{
+			@Override
+			public void focusLost(FocusEvent arg0)
+			{
+				// everything is okay with the new value name, lets rename everything
+				final ArrayList<String> newValues = MapXMLHelper.s_gameSettings.get(m_currentRowName);
+				newValues.set(1, (String) m_tEditable.getSelectedItem());
+			}
+			
+			@Override
+			public void focusGained(FocusEvent arg0)
+			{
+			}
+		});
+		
+		dimension = m_tMinNumber.getPreferredSize();
+		dimension.width = INPUT_FIELD_SIZE_SMALL;
+		m_tMinNumber.setPreferredSize(dimension);
+		m_tMinNumber.setEnabled(!m_isBoolean);
+		m_tMinNumber.addFocusListener(new FocusListener()
+		{
+			
+			@Override
+			public void focusLost(FocusEvent arg0)
+			{
+				String inputText = m_tMinNumber.getText().trim();
+				try
+				{
+					final Integer newValue = Integer.parseInt(inputText);
+					if (newValue < 0)
+						throw new NumberFormatException();
+					final Triple<String, String, Integer> oldTriple = MapXMLHelper.s_playerSequence.get(m_currentRowName);
+					MapXMLHelper.s_playerSequence.put(m_currentRowName, new Triple<String, String, Integer>(oldTriple.getFirst(), oldTriple.getSecond(), newValue));
+				} catch (NumberFormatException e)
+				{
+					m_tMinNumber.setText("0");
+					JOptionPane.showMessageDialog(stepActionPanel, "'" + inputText + "' is no integer value.", "Input error", JOptionPane.ERROR_MESSAGE);
+					parentRowPanel.setDataIsConsistent(false);
+					SwingUtilities.invokeLater(new Runnable()
+					{
+						@Override
+						public void run()
+						{
+							m_tMinNumber.updateUI();
+							m_tMinNumber.requestFocus();
+							m_tMinNumber.selectAll();
+						}
+					});
+					return;
+				}
+				parentRowPanel.setDataIsConsistent(true);
+			}
+			
+			@Override
+			public void focusGained(FocusEvent arg0)
+			{
+				m_tMinNumber.selectAll();
+			}
+		});
+		
+		dimension = m_tMaxNumber.getPreferredSize();
+		dimension.width = INPUT_FIELD_SIZE_SMALL;
+		m_tMaxNumber.setPreferredSize(dimension);
+		m_tMaxNumber.setEnabled(!m_isBoolean);
+		m_tMaxNumber.addFocusListener(new FocusListener()
+		{
+			
+			@Override
+			public void focusLost(FocusEvent arg0)
+			{
+				String inputText = m_tMaxNumber.getText().trim();
+				try
+				{
+					final Integer newValue = Integer.parseInt(inputText);
+					if (newValue < 0)
+						throw new NumberFormatException();
+					final Triple<String, String, Integer> oldTriple = MapXMLHelper.s_playerSequence.get(m_currentRowName);
+					MapXMLHelper.s_playerSequence.put(m_currentRowName, new Triple<String, String, Integer>(oldTriple.getFirst(), oldTriple.getSecond(), newValue));
+				} catch (NumberFormatException e)
+				{
+					m_tMaxNumber.setText("0");
+					JOptionPane.showMessageDialog(stepActionPanel, "'" + inputText + "' is no integer value.", "Input error", JOptionPane.ERROR_MESSAGE);
+					parentRowPanel.setDataIsConsistent(false);
+					SwingUtilities.invokeLater(new Runnable()

```

Comment:
will do, never used that before 

**@DanVanAtta** Let me answer to your previous comments here, as I cannot respond to them where they are posted:

**1. Purpose of the PR**: Vergryn and I were trying to convert the existing tool to Java to have it maintainable in the same project. We failed so far and this is why I was trying to rebuild it by hand. The amount of work was increasing and my goal was just to have a (working) starting point from which we take it further, so I decided to hand it in. Then we changed to github and I needed to get adjusted to the process here as well and didn't find the time at the beginning. This leads me to:

**2. How would I like to proceed**: I would like to see the work not getting wasted and would be willing to help for that. My suggestion would be to prepare the PR to reach a state which is acceptable for the team while keeping in mind that also a TODO statement is a possible quick solution.
Currently, I am trying to go over all your comments and do adjustments are continue debating. Would that strategy work for you guys?